nutils/function.py

```diff
@@ -2621,11 +2621,17 @@ def __init__(self, ndofs: int, nelems: int, index: Array, coords: Array) -> None
     self.coords = coords
     super().__init__((ndofs,), float)
 
+    self.arg_dofs, self.arg_coeffs = [f.optimized_for_numpy
+      for f in self.f_dofs_coeffs(evaluable.Argument('_index', shape=(), dtype=int))]
```

Comment:
What do you think of making `f_dofs_coeffs` an `evaluable.Tuple`? Then we can write

```python
self.arg_dofs, self.arg_coeffs = self.f_dofs_coeffs(evaluable.Argument('_index', shape=(), dtype=int)).optimized_for_numpy
```

Suggestion for readability: Introduce a variable for the `Argument('_index', ...)`.

Suggestion: Make `self.arg_dofs` and `self.arg_coeffs` private.

Reply:
Well, that adds a lot of cruft to other places, so forget about the `evaluable.Tuple`. But maybe with a variable for `Argument` we can get rid of the newline in the list comprehension?

Reply:
done.